packages/frontend/src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,190 @@
+import { Colors } from '@/styles/colors'
+import styled from 'styled-components'
+import { CrossIcon } from '../icons'
+
+interface StepContent {
+  name: string
+  description?: string
+}
+
+type StepDescription = 'default' | 'failed'
+
+type Step = Record<StepDescription, StepContent>
+
+type Steps = Step[]
+
+type StepType = 'neutral' | 'success' | 'failure'
+
+interface StepperProps {
+  steps: Steps
+  currentStep: number
+  isFailed: boolean
+}
+
+export const TransactionStepper = ({ currentStep, isFailed, steps }: StepperProps) => (
+  <StepperList>
+    {steps.map((item, index) => {
+      const isCurrent = index === currentStep
+      const status = isCurrent ? 'current' : index < currentStep ? 'completed' : 'next'
+      const { step, type } = pickStepVersion(item, isFailed && isCurrent, isCurrent)
+      return <StepperListItem key={index} step={step} status={status} type={type} />
+    })}
+  </StepperList>
+)
+
+function pickStepVersion(item: Step, IsFailed: boolean, isCurrent: boolean) {
+  const [step, type]: [StepContent, StepType] =
+    item.failed && IsFailed ? [item.failed, 'failure'] : [item.default, isCurrent ? 'neutral' : 'success']
+  return { step, type }
+}
+
+type StepStatus = 'current' | 'completed' | 'next'
+
+interface ListItemProps {
+  step: StepContent
+  status: StepStatus
+  type: StepType
+}
+
+const StepperListItem = ({ step, status, type }: ListItemProps) => (
+  <StepperListItemContainer type={type} status={status}>
+    <StepperBullet type={type} status={status}>
+      {type === 'failure' && <CrossIcon size={16} color={Colors.RedDark} />}
+    </StepperBullet>
+    <StepperItemName next={status === 'next'}>{step.name}</StepperItemName>
+    <StepperItemDescription current={status === 'current'}>{step.description}</StepperItemDescription>
+  </StepperListItemContainer>
+)
+
+const StepperList = styled.ul`
+  display: flex;
+  flex-direction: column;
+  padding: 0;
+  margin: 0;
+`
+
+function getItemColor(props: DisplayTypeProps) {
+  switch (props.status) {
+    case 'current':
+      switch (props.type) {
+        case 'neutral':
+          return Colors.Black
+        case 'success':
+          return Colors.Black
+        case 'failure':
+          return Colors.RedDark
+      }
```

Comment:
Would look better if we use record for that (no nested switch statements and clear return statement), what do you think?
```suggestion
      return currentItemToColor[props.type]
    default:
      return Colors.Black
  }
}

const currentItemToColor: Record<StepType, string> = {
  neutral: Colors.Black,
  success: Colors.Black,
  failure: Colors.RedDark
}
```